packages/back-end/src/scim/groups/createGroup.ts

```diff
@@ -0,0 +1,59 @@
+import { Response } from "express";
+import { createTeam } from "../../models/TeamModel";
+import {
+  ScimError,
+  ScimGroup,
+  ScimGroupPostRequest,
+} from "../../../types/scim";
+import { addMemberToTeam } from "../../services/organizations";
+
+const DEFAULT_TEAM_PERMISSIONS = {
+  role: "collaborator",
+  limitAccessByEnvironment: false,
+  environments: [],
+};
+
+export async function createGroup(
+  req: ScimGroupPostRequest,
+  res: Response
+): Promise<Response<ScimGroup | ScimError>> {
+  const { displayName, members } = req.body;
+
+  const org = req.organization;
+
+  try {
+    const group = await createTeam({
```

Comment:
Whenever you try to push a group, Okta does a get request to see if a group with the same name already exists and gives you the option to link the groups so I think we don't have to handle for that explicitly. I do think it's a good idea to enforce making team names unique though and I will add that check. I agree that we probably need to either have a `managedByIdp` boolean value for each team or we could just have a binary choice where admins either allow teams to be created within GrowthBook or within their idP. I'm kind of leaning towards the latter because teams is a new feature and I can't really think of why an org would need both to be possible. Seems like unnecessary confusion, but let me know if you think of any use case that would require that functionality

Reply:
Actually on second thought, may be good to support both non idp managed teams and idp managed teams in the case that the org starts off not using an idp and then wants to transition to using one. If its one or the other then the org would probably need to delete their existing GrowthBook managed teams before setting up team management with their idp and that seems less than ideal. May need to add a toggle to enable idP group sync in the UI

Reply:
Oh! Something to note is that Okta doesn't allow using the same Groups for assignment and for push groups so its worth noting that limitation somewhere